test/Microsoft.ML.Tests/AnomalyDetectionTests.cs

```diff
@@ -29,33 +22,57 @@ public AnomalyDetectionTests(ITestOutputHelper output) : base(output)
         [Fact]
         public void RandomizedPcaTrainerBaselineTest()
         {
-            var mlContext = new MLContext(seed: 1, conc: 1);
-            string featureColumn = "NumericFeatures";
-
             var reader = new TextLoader(Env, new TextLoader.Options()
             {
-                HasHeader = true,
+                HasHeader = false,
                 Separator = "\t",
                 Columns = new[]
                 {
                     new TextLoader.Column("Label", DataKind.R4, 0),
-                    new TextLoader.Column(featureColumn, DataKind.R4, new [] { new TextLoader.Range(1, 784) })
+                    new TextLoader.Column(DefaultColumnNames.Features, DataKind.R4, new [] { new TextLoader.Range(1, 784) })
                 }
             });
 
             var trainData = reader.Read(GetDataPath(TestDatasets.mnistOneClass.trainFilename));
```

Comment:
if you create the reader from the mlcontext you can make it a part of the pipeline.

Reply:
Done